streaming/vision/convert/coco.py

```diff
@@ -181,15 +181,15 @@ def each(dataset: _COCODetection, shuffle: bool) -> Iterable[Dict[str, bytes]]:
         img_data = dataset.images[img_id]
         img_basename = img_data[0]
         img_filename = os.path.join(dataset.img_folder, img_basename)
-        img_bytes = open(img_filename, 'rb').read()
+        img = Image.open(img_filename)
```

Comment:
let's keep this as bytes so as to not decode and then reencode images (passing bytes should work here)

Reply:
It has to decode anyways [here](https://github.com/mosaicml/streaming/blob/main/streaming/base/format/mds/encodings.py#L111) right if encoding format is `jpeg`!!  Then why passing it as `bytes` ? If we pass it as `bytes` then I see that the encoding is getting skipped [here](https://github.com/mosaicml/streaming/blob/main/streaming/base/format/mds/encodings.py#L193). Wondering that's hidden since I added encoding format as `jpeg`.  So I kept encoding as `jpeg` and I put image as of type `PIL.Image` so that proper encoding and decode method gets called.

Reply:
during conversion, these images are already encoded as jpeg, so we take the bytes from the individual image files and directly write them into our mds shards

we avoid decoding to pil image and reencoding back to bytes in the conversion script because it's (a) slower and (b) jpeg is lossy, so the data will in all likelihood be changed by such a process

skipping reencoding jpeg is why i added that magic where it accepts raw bytes in the image encoding method

if i am understanding you correctly

Reply:
Yes Make Sense. Thanks for the explanation. I moved back to bytes in https://github.com/mosaicml/streaming/pull/13/commits/20fab46995190a3fbd24ad7c4f4e08249de05091.